### PR TITLE
Add timestamp token

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

There is a chance to send logs from different users in one batch. To avoid that we need to filter stored logs by token's valid time.

## Related PRs or issues

[AB#58503](https://msmobilecenter.visualstudio.com/web/wi.aspx?pcguid=27bf89de-9697-418a-accf-1aa125b22914&id=58503)

### CODE REVIEW

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java [hunk 1]
```diff
@@ -217,7 +228,10 @@ public boolean onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
                 if (oldVersion < VERSION_TARGET_KEY) {
                     db.execSQL("ALTER TABLE " + TABLE + " ADD COLUMN `" + COLUMN_TARGET_KEY + "` TEXT");
                 }
-                db.execSQL("ALTER TABLE " + TABLE + " ADD COLUMN `" + COLUMN_PRIORITY + "` INTEGER DEFAULT " + PERSISTENCE_NORMAL);
+                if (oldVersion < VERSION_PRIORITY_KEY) {
+                    db.execSQL("ALTER TABLE " + TABLE + " ADD COLUMN `" + COLUMN_PRIORITY + "` INTEGER DEFAULT " + PERSISTENCE_NORMAL);
+                }
+                db.execSQL("ALTER TABLE " + TABLE + " ADD COLUMN `" + COLUMN_TIMESTAMP + "` INTEGER DEFAULT " + PERSISTENCE_NORMAL);
```
Comment (MatkovIvan) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java]: Incorrect usage of `PERSISTENCE_NORMAL`

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/context/AuthTokenContext.java [hunk 2]
```diff
@@ -7,15 +7,24 @@
```
Comment (MatkovIvan) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/context/AuthTokenContext.java]: It looks like this file comes from another PR. Please rebase to `feature/identity`

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java [hunk 3]
```diff
@@ -105,7 +106,7 @@ public void onCreate(SQLiteDatabase db) {
                     if (val instanceof Double || val instanceof Float) {
                         sql.append("REAL");
                     } else if (val instanceof Number || val instanceof Boolean) {
-                        sql.append("INTEGER");
+                        sql.append("INTEGER DEFAULT " + PERSISTENCE_NORMAL);
```
Comment (MatkovIvan) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java]: Incorrect change. Please revert

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/persistence/DatabasePersistenceAndroidTest.java [hunk 4]
```diff
@@ -1120,6 +1121,7 @@ public void upgradeFromVersion1to4() throws PersistenceException, JSONException
             /* Check priority migration. */
             ContentValues values = getContentValues(persistence, "test");
             assertEquals((Integer) PERSISTENCE_NORMAL, values.getAsInteger(DatabasePersistence.COLUMN_PRIORITY));
+//            assertEquals((Long) 0l, values.getAsLong(DatabasePersistence.COLUMN_TIMESTAMP));
```
Comment (jwallra) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/persistence/DatabasePersistenceAndroidTest.java]: Remove commented line

**Comment by guperrot** [comment]
Please do not work on a fork for PRs targeting feature/identity as there is a private dependency for now. @MatkovIvan please help for setting up main repo push and MSAL repo credentials.

**Pull request closed without merging**